src/action.rs

```diff
@@ -181,18 +199,45 @@ impl Action<'_> {
                 let (buf_2, filename_2) = dumper.get_from_id(&path_2, filename_2)?;
 
                 match (FileType::from_buf(&buf_1), FileType::from_buf(&buf_2)) {
-                    (FileType::Elf, FileType::Elf) => {
-                        dumper.two_elfs(buf_1, filename_1, buf_2, filename_2)
-                    }
+                    (FileType::Elf, FileType::Elf) => dumper.two_elfs(
+                        move || {
+                            linux::elf::ElfInfo::new(
+                                &buf_1,
+                                filename_1,
+                                linux::elf::Platform::Linux,
+                            )
+                        },
+                        move || {
+                            linux::elf::ElfInfo::new(
+                                &buf_2,
+                                filename_2,
+                                linux::elf::Platform::Linux,
+                            )
+                        },
+                    ),
                     (FileType::Pdb, FileType::Pe) => {
                         dumper.pdb_pe(&buf_1, filename_1, &buf_2, path_2, filename_2)
                     }
                     (FileType::Pe, FileType::Pdb) => {
                         dumper.pdb_pe(&buf_2, filename_2, &buf_1, path_1, filename_1)
                     }
+                    (FileType::Macho, FileType::Macho) => {
+                        let arch = Arch::from_str(dumper.arch).map_err(|e| e.compat())?;
+                        dumper.two_elfs(
+                            move || mac::macho::MachoInfo::new(&buf_1, filename_1, arch),
+                            move || mac::macho::MachoInfo::new(&buf_2, filename_2, arch),
+                        )
+                    }
                     _ => Err("Invalid files: must be two elf or a pdb and a pe".into()),
                 }
             }
+            Self::ListArch => {
+                let buf = utils::read_file(&path_1);
+                mac::macho::MachoInfo::print_architectures(&buf, filename_1)?;
+
+                let buf = utils::read_file(&path_2);
+                mac::macho::MachoInfo::print_architectures(&buf, filename_2)
+            }
```

Comment:
Lots of indentation here, better to move this to a separate function and possibly split the match arms in smaller functions.

Reply:
I'll fix that in another patch to better handle multiple file in entry.